shogun/structure/DualLibQPBMSOSVM.cpp

```diff
@@ -69,6 +61,17 @@ void DualLibQPBMSOSVM::init()
 	SG_ADD(&m_Tmax, "m_Tmax", "Parameter Tmax", ParameterProperties::HYPER);
 	SG_ADD(&m_cp_models, "m_cp_models", "Number of cutting plane models");
 
+	// TODO(gf712) should be replaced with an extension of Constraint class
+	// which has a customisation point with lambdas, rather than write a whole struct
+	add_callback_function("w", [&](){
+		if (auto dim = m_model->get_dim(); get_w().vlen != dim)
+		{
+			set_w(SGVector<float64_t>(dim));
+			get_w().zero();
+			error("Dimension of the initial solution must match the model's dimension!\n");
```

Comment:
Could you update this to something like "Dimension of the initial solution %d must match the model's dimension %d"?